apps/stepcounter/stepcounter.star

```diff
@@ -0,0 +1,420 @@
+GOOGLE_CLIENT_ID = secret.decrypt(CLIENT_ID_HASH)
+GOOGLE_CLIENT_SECRET = secret.decrypt(CLIENT_SECRET_HASH)
+
+def main(config):
+    # Grab Secrets from paraneters if running in Render mode
+
+    if GOOGLE_CLIENT_ID:
+        client_id = GOOGLE_CLIENT_ID
+    else:
+        client_id = config.get("client_id")
+
+    if GOOGLE_CLIENT_SECRET:
+        client_secret = GOOGLE_CLIENT_SECRET
+    else:
+        client_secret = config.get("client_secret")
+
+    # Google Project Required Info for OAuth
+    # Note to Developers: You need to create your own project in the Google developers console in order to obtain a Client ID and Client Secret.
+
+    # Get the Refresh Token - from Schema if running in Serve/Production mode or via Params when running in Render mode.
+
+    GOOGLEFIT_REFRESH_TOKEN = config.get("auth") or config.get("googlefit_refresh_token")
+
+    # Stepcount cannot be retrieved if a Refresh Token is not present
+    # Render "Unauthorized" in that case
+
+    if not GOOGLEFIT_REFRESH_TOKEN:
+        return render.Root(
+            child = render.WrappedText("Not Authorized (Stepcounter)"),
+        )
+    else:
+        GOOGLEFIT_OAUTH_TOKEN = cache.get(GOOGLEFIT_REFRESH_TOKEN)
+
+    # Get New Access token if Cache expired
+    if not GOOGLEFIT_OAUTH_TOKEN:
+        print("Fetching Access Token")
+
+        # This is the JSON format for supplying an OAUTH refresh token to receive a new Authorization Token
+
+        GOOGLE_OAUTH_TOKEN_REFRESH_BODY = {
+            "client_secret": client_secret,
+            "grant_type": "refresh_token",
+            "refresh_token": GOOGLEFIT_REFRESH_TOKEN,
+            "client_id": client_id,
+        }
+
+        # Make the Google Oauth API call - POST - to exchange Refresh token for Auth Token
+        # For some reason this works just fine without the JSON command header (is that header really needed for any of the calls?)
+
+        refresh = http.post(GOOGLE_OAUTH_TOKEN_REFRESH_URL, json_body = GOOGLE_OAUTH_TOKEN_REFRESH_BODY)
+        if refresh.status_code != 200:
+            fail("Google OAUTH TOKEN API request failed with status:", refresh.json())
+
+        # Grab new Oauthtoken from the Google Token service, format for Data Aggregation API call.
+        GOOGLEFIT_OAUTH_TOKEN = "Bearer {}".format(refresh.json()["access_token"])
+        cache.set(GOOGLEFIT_REFRESH_TOKEN, GOOGLEFIT_OAUTH_TOKEN, ttl_seconds = int(refresh.json()["expires_in"] - 30))
+
+    else:
+        GOOGLEFIT_OAUTH_TOKEN = cache.get(GOOGLEFIT_REFRESH_TOKEN)
+
+    # Header to Specify JSON format for the FIT API Data Aggregation
+    GOOGLEFIT_POST_HEADERS = {
+        "Content-type": "application/json",
+        "Authorization": GOOGLEFIT_OAUTH_TOKEN,
+    }
+
+    # LOOKS LIKE A BUG HERE (Timezone)?  How does the Tidbyt Server pick the timezone?   Need to figure that out.  There is some discussion on Discord - search "Time Zone"
```

Comment:
You can add a location object to make sure you're getting the desired timezone. Checkout this example: https://github.com/tidbyt/pixlet/blob/main/examples/sunrise.star#L35

Reply:
Thanks @betterengineering I will get this done in a future PR.